pkg/models/commandvalue_test.go

```diff
@@ -569,3 +571,146 @@ func TestNewFloat64Value(t *testing.T) {
+	isValidated bool      // internal member used for validation check
+}
+
+// Accepts a mockStructB and returns a CBOR encoded byte array
+func encodeMockEvent(e mockStructB) ([]byte, error) {
+	var handle codec.CborHandle
+	var byteBuffer = make([]byte, 0, 64)
+	enc := codec.NewEncoderBytes(&byteBuffer, &handle)
+	err := enc.Encode(e)
+	if err != nil {
+		return []byte{}, err
+	}
+	return byteBuffer, nil
+}
+
+// Accepts a CBOR encoded byte array and returns mockStructB
+func decodeMockEvent(encodedData []byte) (mockStructB, error) {
+	var err error
+	evt := mockStructB{}
+	err = decodeBinaryValue(bytes.NewReader(encodedData), &evt)
+	return evt, err
+}
+
+func createMockPayload(size int) ([]byte, error) {
+	token := make([]byte, size)
+	_, err := rand.Read(token)
+	if err != nil {
+		return nil, err
+	}
+	return token, nil
+}
+
+// Test NewBinaryValue function and associated methods for binary encode/decode.
+func TestNewBinaryValue(t *testing.T) {
+	var origin int64 = time.Now().UnixNano() / int64(time.Millisecond)
+	// assign instance of mockStructB as a CBOR encoded CommandValue payload
+	var mock1 mockStructB
+	var mock2 mockStructB
+	mock1.Device = "Device01234567890"
+	mock1.Created = origin
+	mock1.ID = "MyStringIdentifier"
+	mock1.Modified = origin + 123
+	mock1.Pushed = 12345
+	// To extend coverage cborMock1 becomes encoded byte array.
+	// We will then confirm CommandValue particulars of binary payload are valid
+	cborMock1, err := encodeMockEvent(mock1)
+	if err != nil {
+		t.Errorf("NewBinaryValue: Error encoding struct as binary value")
+	}
+	cv, errAssign := NewBinaryValue("resource", origin, cborMock1)
+	if errAssign != nil {
+		t.Errorf("NewBinaryValue: Error invoking NewBinaryValue [%v]", errAssign)
+	}
+	// Confirm CommandValue particulars
+	if cv.Type != Binary {
+		t.Errorf("Expected Binary type! invalid Type: %v", cv.Type)
+	}
+	if cv.Origin != origin {
+		t.Errorf("Expected matching value! invalid Origin: %d != %d", cv.Origin, origin)
+	}
+	// Populate cv.BinValue as CBOR encoded CommandValue and decode back
+	err1 := encodeBinaryValue(cv, mock1)
+	if err1 != nil {
+		t.Errorf("NewBinaryValue: Error encoding binary value")
+	}
+	// decode the original encoded value, compare original raw struct to decoded mock2 and mock3
+	var err2 error
+	mock2, err2 = decodeMockEvent(cborMock1)
+	if err2 != nil {
+		t.Errorf("NewBinaryValue: Error decoding binary value")
+	}
+	if mock1 != mock2 { // deep compare of mock struct
+		t.Errorf("NewBinaryValue: cv.BinValue: %v doesn't match value: %v", mock2, mock1)
+	}
+	mock3, err3 := decodeMockEvent(cv.BinValue)
+	if err3 != nil {
+		t.Errorf("NewBinaryValue: Error decoding binary value")
+	}
+	if mock2 != mock3 { // deep compare of mock struct
+		t.Errorf("NewBinaryValue: cv.BinValue: %v doesn't match value: %v", mock2, mock3)
+	}
+}
+
+// Test NewBinaryValueConstraints function and associated methods for binary encode/decode.
+func NewBinaryValueWithinConstraints(t *testing.T) {
```

Comment:
This function is never used. I've commented it out and successfully run both a `make build` and `make test`. Can it be removed? If the intent is to make this a test, try changing the name to `TestNewBinaryValueWithinConstraints`